Dockerfile

```diff
@@ -110,21 +110,48 @@ RUN file -b   /bin/invoicer
 RUN du        /bin/invoicer
 
 
+# This stage is used to generate /etc/{group,passwd,shadow} files & avoid RUN-ing commands in the `final` layer,
+#   which would break cross-compiled images.
+FROM alpine:3.11 AS perms
+
+# The most generic linux-native name for the user I could quickly come up with 🤷🏻‍♂️
+ARG USER=tux
+ARG DIR=/data/
+
+# NOTE: Default GID == UID == 1000
+RUN adduser --disabled-password \
+            --home ${DIR} \
+            --gecos "" \
+            ${USER}
+
+
 # This is a final stage, destined to be distributed, if successful
 FROM alpine:3.11 AS final
 
+ARG USER=tux
+ARG DIR=/data/
+
 # Hai 👋
 LABEL maintainer="Damian Mee (@meeDamian)"
 
+# Copy only the relevant parts from the `perms` image
+COPY  --from=perms  /etc/group   /etc/
+COPY  --from=perms  /etc/passwd  /etc/
+COPY  --from=perms  /etc/shadow  /etc/
+
 # Copy the binary from the cross-check stage
 COPY  --from=cross-check  /bin/invoicer  /bin/
 
-# Expose the volume to communicate config, log, etc through
-VOLUME /root/.lncm
+# Expose the volume to communicate config, log, etc through (default: /data/)
+VOLUME ${DIR}
 
-# Expose Invoicer port
+# Expose port Invoicer listens on
 EXPOSE 8080
 
+USER ${USER}
+WORKDIR ${DIR}
+
 # Specify the start command and entrypoint as the invoicer daemon.
-ENTRYPOINT ["/bin/invoicer"]
+ENTRYPOINT ["invoicer"]
 
+CMD ["-config", "/data/invoicer.conf"]
```

Comment:
Why is `-config /data/invoicer.conf` now hardcoded?

If I'm understanding correctly, it defaults to looking in `$HOME/.lncm/` and you would previously just mount that volume with an `invoicer.conf`.

That seems like abetter solution to me. I would avoid default params where possible because people might want to override them.

For example I might want to run:

```
$ ls
invoicer.conf
invoicer_testnet.conf
$ docker run -v $PWD:/data/.lncm/ lncm/invoicer
// or
$ docker run -v $PWD:/data/.lncm/ lncm/invoicer -config /data/invoicer_testnet.conf
```

Currently the last option will actually be running `invoicer -config /data/invoicer.conf -config /data/invoicer_testnet.conf` inside the container. I would imagine (haven't checked) this would probably still work if `invoicer` takes the last argument as the highest priority but it's still kind of confusing.

Reply:
```sh
$ cat Dockerfile
FROM alpine:3.11

CMD ["aaa"]
ENTRYPOINT ["echo"]
$ docker run --rm -it echo-test
aaa
$ docker run --rm -it echo-test bbb
bbb
```

`CMD` is used only if nothing is provided by the user.

The only flags exposed by invoicer are `-config`, and `-version`.  So by providing either, the user either overrides the location of the config file, or asks for invoicer version.

Reply:
I'm still not quite happy with hardcoding `"/data/"` there, as it's a variable `DIR` in the file, but I can't quite figure out an easy way to have it expanded in the `CMD` command directive.

Reply:
Ahh, was not aware `CMD` was dropped in the event of user supplied arguments!